### DISCUSSION TITLE
[SP-API] [Issue] Authorization of access to PII information to get shipping address information.

### DISCUSSION THREAD

**Discussion by WiFeng** [amzn/selling-partner-api-models#3085] [General]
1.  https://github.com/amzn/selling-partner-api-docs/issues/394
2. https://github.com/amzn/selling-partner-api-docs/issues/1217
3. https://github.com/amzn/selling-partner-api-docs/issues/1331
4. https://github.com/amzn/selling-partner-api-docs/issues/1529
5. https://github.com/amzn/selling-partner-api-docs/issues/1709
6. https://github.com/amzn/selling-partner-api-docs/issues/1710
7. https://github.com/amzn/selling-partner-api-docs/issues/1730
8. https://github.com/amzn/selling-partner-api-docs/issues/1737
9. https://github.com/amzn/selling-partner-api-docs/issues/1938

@9z  @misterakko @valigara  @petyayuliyanova
Have you all successfully obtained the PII data now? Did you try other ways later?

@Javadebi  @DDDDDID  @optikalefx  
How did you apply for PII permission? Can you help give us some advice? Does Amazon actually evaluate the size of the company, the team members in the security department, and the data protection plan during this approval process?

Looking forward to your reply

**Comment by optikalefx** [comment]
@WiFeng go here https://sellercentral.amazon.com/sellingpartner/developerconsole and fill out the form.

**Comment by WiFeng** [comment]
@optikalefx  Thanks for your reply. 

 I submitted my application a few days ago and it was rejected. As follow:

First reply
> We have completed our assessment and have determined that you do not meet the requirements for Restricted SP-API. To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon.

Sencond reply
> We have reviewed the information you provided and determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time.

> We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies.

Third reply

> As previously mentioned, we have determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time.

> We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies.

>  Please note that this decision is final and we may not respond to further inquiries about this.

Then, yestoday at this github project I found much more people meet the same problem. I would like to ask if there is a public solution now ?

**Comment by c-prasanth** [comment]
I have applied and i to got same response
the response is as follows.

We have completed our assessment and have determined that you do not meet the requirements for Restricted SP-API. To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon.
my case number is 8126736572

**Comment by optikalefx** [comment]
Like I’ve said in previous posts, if your not at least a 200 person company
with a full IT department, infosec department, certifications, compliance,
development standards, devOps etc - then there is a slim chance you’re
going to pass the security audit. It’s really only available to companies
that have a significant investment in security.

- Sean

**Reply by dwoj714** [reply]
I'm a developer from a small company, roughly 20 people, and we want to enable our systems to autonomously pull orders for our customers who are, or will be, sellers on Amazon.
We've already been rejected for the restricted roles, and we've been under the assumption that we'd be able to as long as we improve our "launch readiness" and address the other issues mentioned in their copy/pasted rejection letter.
Is there any hope for us, or does Amazon just not touch small companies with a 10-foot pole when it comes to restricted data?

**Reply by Hinten** [reply]
> Like I’ve said in previous posts, if your not at least a 200 person company with a full IT department, infosec department, certifications, compliance, development standards, devOps etc - then there is a slim chance you’re going to pass the security audit. It’s really only available to companies that have a significant investment in security. - Sean
> […](#)
> On Sat, Nov 5, 2022 at 12:25 AM c-prasanth ***@***.***> wrote: I have applied and i to got same response the response is as follows. We have completed our assessment and have determined that you do not meet the requirements for Restricted SP-API. To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon. my case number is 8126736572 — Reply to this email directly, view it on GitHub <[#2712 (comment)](https://github.com/amzn/selling-partner-api-docs/discussions/2712#discussioncomment-4062016)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAHCJJLA4AGWK3SFVN4UOUDWGXOUXANCNFSM6AAAAAARXXV5UA> . You are receiving this because you were mentioned.Message ID: <amzn/selling-partner-api-docs/repo-discussions/2712/comments/4062016@ github.com>

If this is true, then why Amazon dont put this as a criteria in the docs?

**Reply by ftf1218** [reply]
我也有同样的问题，如果这是真的，那么为什么亚马逊不将其作为申请表格文档中的标准呢？

**Comment by shuichisunday** [comment]
@optikalefx @WiFeng 

I have submitted the application for 4 times since June, and all were rejected.
We are a 186-person company, and we have p-mark. Do you think we need ISMS and definitely more than 200 people?

**Comment by optikalefx** [comment]
Have they told you any reasons for the rejection? You’re big enough that
you should have the proper data protection capabilities.

I could check your answers to the questions if you’d like.

**Comment by shuichisunday** [comment]
@optikalefx 

Here are the comments for the rejection.

First rejection on 6/30/2022 for application on 6/26/2022
> We have completed our assessment and have determined that you do not meet the requirements for Restricted SP-API. To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon. No changes have been made to your SP-API access at this time.

Secund rejection on 9/6/2022 for application on 8/30/2022

> We have reviewed the information you provided and determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time.  We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies.  If you decide to build a publicly available application with the SP-API roles you already have access to, it is required to be listed in the Amazon Selling Partner Network. 

Third rejection on 10/21/2022 for application on 10/21/2022

> We have reviewed the information you provided and determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time. We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies. If you decide to build a publicly available application with the SP-API roles you already have access to, it is required to be listed in the Amazon Selling Partner Network. 

Forth rejection on 12/6/2022 for application on 11/24/2022

> As previously mentioned, we have determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time. We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies. Please note that this decision is final and we may not respond to further inquiries about this.

Thank you for your help.

**Reply by optikalefx** [reply]
These responses are fishy. Something doesn't feel like it's adding up right. Amazon always responds with the exact section and question for you re-work. They refer to the numbered sections in this document https://sellercentral.amazon.com/mws/static/policy?documentType=DPP&locale=en_US for each question you need to re-submit.

Their answers to you, sound like there is something else going on with your relationship with Amazon. I'm not sure what that could be. From the sounds of it, it sounds like Amazon isn't able to determine that your company is legit. 

```
launch readiness, services offered that can be viewed on a public facing website
```

Do you want to share your website and a sample of the answers you gave to Amazon?

**Comment by hjsdygtfshdg** [comment]
Hey,

Does anyone have any clue why amazon has f-ed-up this so much and can anything be done? (other than subscribing to some data aggregating app that has passed PII and have them deliver data - good job Amazon, now the PII will end up in 3rd party and in my DB, especially if I have to pay-per-request)

I understand PII needs to be safe and requirements like they have set up could be enforced for apps allowing anyone to authorise (public apps), but why would PII be hidden from my own account to my own app (Self-Authorisations) - I can open web and copy paste the information, download CSV order report and have the information, well I could just run headless chrome and fetch all the information on server this way - why would you make it difficult?

I would like to get this info for one simple reason - buy and print label, prepare invoices, upload them and update statuses, now I'm copy-pasting everything manually and it gets annoying real fast.

We are a two-man army - so there is no crazy team of hundreds developing some software, analysing permissions and logs 24/7 or accessing the PII. I know code and I know security, both I can prove with a fancy paper, though that paper is not much in real world.  Anyhow we host everything in AWS, so basic security is covered unless our MFA devices and computer get accessed, but in that case seller central can be accessed, so no difference.

In the end we don't even need to store PII, they can keep it and every time I will need I will request it - no problem works for me, I can generate PDF, buy shipping label, send both to printer, update statuses and delete - store order id and tracking. We actually need PII only once, unless we are replacing damaged shipment.

Currently thinking about some automatic order report downloader and processor, but I can't believe that this is necessary, so hopefully I have missed something.

Thanks for any input.

**Comment by Moeschie** [comment]
Hello everyone,

after having the same issue for months and no feedback on github, Amazon Support and paid Amazon account managers. I have reached out to the German authorities "Bundeskartellamt" and it turned out that they are very much interested in this case as they do not have any relatable issue with Amazon in the sector, only other parts were they are proceeding against Amazon
(https://www.bundeskartellamt.de/SharedDocs/Meldung/DE/Pressemitteilungen/2022/14_11_2022_Amazon_19a.html & https://www.bundeskartellamt.de/SharedDocs/Meldung/EN/Pressemitteilungen/2018/29_11_2018_Verfahrenseinleitung_Amazon.html?nn=3591568).

I have had multiple meetings with the Bundeskartellamt and explained the issues on why the application rejections are so problematic as they do not contain an actual reason. 

I have created a document containing the full application process with questions and the rejections of Amazon. The contact of the Bundeskartellamt has asked if there are more EU or even German companies that they can contact regarding this issue, that’s why I’m posting this here. If you are interested, I would add your contacts to the Document that the Bundeskartellamt can reach out to you. This is all anonym towards Amazon and would help them to build a more solid foundation for their case.
Cheers everyone and a happy new year.

**Reply by nfzv** [reply]
There are numerous cases recorded by the following company regarding this and various other issues, please leave the following email: solution.latvia@gmail.com Thanks!

**Comment by Lemodo20** [comment]
I have also been trying to answer these strange questions for a few weeks. I always fail with a few questions. How about we throw our answers together. Maybe someone else will have the correct answer for Amazon. I think they work stubbornly according to a list and if the right keywords are in there, then you get access.
Otherwise, I would also be on board with the Federal Cartel Office. I am also from DE

**Reply by misterakko** [reply]
I have filled in the form more times than I can count in the last couple of years... It's not that some answers of mine are "wrong", it's just that they don't have the will to go ahead. Every time they say that 99% of what I say is OK, but there's one thing they don't like. So I fix that issue (being more precise and explaining better how we do things) and the process repeats.
Also, you can't be too detailed, because there is a strict limit on how much text you can write in the provided box.

Three years ago we were handling sales for a huge company (which, unfortunately, asked us for a copy of our software to be used as if they had developed it themselves) and they got the permission almost immediately. Amazon doesn't want to be bothered with many users of their APIs, unless we bring in huge amounts of gross income.

@Moeschie Feel free to add me to your document / give my address to your contact at the authority. I am luca at accomazzi dot net

**Comment by leminhduc2504** [comment]
any update? I have same problem. 

I have submitted the application for 3 times and the comment of rejection is always:  "To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon." 

It even doesn't point out which questions need to be reviewed so I don't have any clue to fix my answer. :'(

**Comment by shuichisunday** [comment]
I have submitted the application again last week, and got the following comments. This is 7th time and no success. 
It looks like they have no intention to give us PII access.

***********************************************************************************************************
As previously mentioned, we have determined that you will not be given access to Restricted SP-API roles based on your application. Your SP-API access remains unchanged at this time.

We evaluate all Public Developer applications that request Restricted SP-API access, for a variety of technical and business criteria including, but not limited to, launch readiness, services offered that can be viewed on a public facing website, and supported geographies.

Please note that this decision is final and we may not respond to further inquiries about this.
***********************************************************************************************************

**Comment by nasenbaer43** [comment]
Same issue: always the same reply and no info what is missing.

Fun part is: we are already validated and have access to PII / restricted access methods with our private vendor app. Since this is a private vendor app we created a new one (public this time) to be able to use it for other clients.

And guess what: no chance.

If anyone from amazon please can have a look? Ticket CASE 8404946402 just got closed. I'm more than happy to supply further data like sellerId, clientId etc from both our accounts. Again we already have access to PII with our old account and used the same answers...

**Reply by baidan** [reply]
I don't understand what the developers at Amazon are doing, did the review team fail on purpose~

**Comment by l20DfX35JnKBfRn** [comment]
Same issue here as well even though we have full access to MWS as a public developer

**Comment by ericahoekerd** [comment]
Same here, we are a Microsoft ISV for 15 years now with 400+ customers and have done a huge investment to rebuild the Business Central app to SP-API, but without the shipping address information, our app is useless and no merchant will ever going to use it.

This is the response we have received:

_We have completed our assessment and have determined that you do not meet the requirements for Restricted SP-API. To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon.

No changes have been made to your SP-API access at this time._

What can we do to get proper access by API?

**Comment by Lemodo20** [comment]
Hi,

 

i tell you, what’s to do. Lie to them until the beams bend.

 

 

Mit freundlichen Grüßen

 

Richard Schülein

Lemodo GmbH
Steindeckerstr. 2

91572 Bechhofen
fon: +49 (0) 9822 / 6099470
fax: +49 (0) 9822 / 60994781

 

Von: Tinx-IT B.V. ***@***.***> 
Gesendet: Freitag, 14. April 2023 11:35
An: amzn/selling-partner-api-docs ***@***.***>
Cc: Lemodo20 ***@***.***>; Comment ***@***.***>
Betreff: Re: [amzn/selling-partner-api-docs] [SP-API] [Issue] Authorization of access to PII information to get shipping address information. (Discussion amzn/selling-partner-api-docs#2712)

 

Same here, we are a Microsoft ISV for years now wthi 300+ customers and have done a huge investment to rebuild the app to SP-API, but without the shipping address information, our app is useless and no merchant will ever going to use it.

This is the response we have received:

_We have completed our assessment and have determined that you do not meet the requirements for Restricted SP-API. To protect Amazon Customers, we only consider sharing Customer data (“Restricted access”) with third party Developers that offer features that are materially different from existing applications. The Developers that meet these criteria must then go through rigorous security reviews with Amazon.

No changes have been made to your SP-API access at this time._

What can we do to get proper access by API?

**Comment by jheathco** [comment]
Same experience here.  Using MWS for years without issue - now unable to get access for the transition.

Should this process be easier to obtain authorization if it's for a private app?

**Reply by milkProvider** [reply]
Im on the same boat. Selenium (*website scraping) is workaround for some time in critical case (especially for private app).

**Reply by jheathco** [reply]
Using mechanize gem here.

**Comment by NA-Dev** [comment]
Amazon needs to be fully transparent about what it takes to be approved. Our company is large, has an IT department and info security, and only wants data for our own sales. Still we cannot be approved. Very frustrating.

**Comment by ericahoekerd** [comment]
hello everyone, we have added a Linkedin post to address this topic which is still actual:

https://www.linkedin.com/feed/update/urn:li:activity:7153061832428064768

It would be great if as many companies can respond to this, thanks a lot!
